config/v1/types_image.go

```diff
@@ -46,6 +46,12 @@ type ImageStatus struct {
 	// For backward compatibility, users can still use OPENSHIFT_DEFAULT_REGISTRY
 	// environment variable but this setting overrides the environment variable.
 	InternalRegistryHostname string `json:"internalRegistryHostname,omitempty"`
+
+	// ExternalRegistryHostname sets the hostname for the default external image
+	// registry. The external hostname should be set only when the image registry
+	// is exposed externally. The value is used in 'publicDockerImageRepository'
+	// field in ImageStreams. The value must be in "hostname[:port]" format.
+	ExternalRegistryHostname string `json:"externalRegistryHostname,omitempty"`
```

Comment:
I would expect this to be a slice.  Why would we constrain ourselves to a single external representation?

Reply:
only one value can be consumed by the product (see the publicDockerImageRepository field in imagestreams), providing a slice means:

1) consumers(such as imagestreams) have to decide which one to use and may pick different ones
2) implying multiple values have meaning when in reality only one does

If we want the ability to expose additional values (ie the admin has setup 15 more domains for their registry) I can see us adding a second field like `ExternalRegistryHostnameAliases []string`

But for the canonical value I think exactly one is appropriate (and it's what we allowed for prior to 4.0 w/o any issue that i'm aware of)

Reply:
@deads2k updated this to be a slice per our conversation, ptal.